proxy/src/auth/backend.rs

```diff
@@ -214,14 +215,48 @@ async fn auth_quirks(
         Some(secret) => secret,
         None => api.get_role_secret(ctx, &info).await?,
     };
+    let (cached_entry, secret) = cached_secret.take_value();
+
+    let secret = match secret {
+        Some(secret) => {
+            // we have validated the endpoint exists, so let's intern it.
+            let endpoint = EndpointIdInt::from(&info.endpoint);
```

Comment:
nit: let's also use `endpoint.normalize()`.

Reply:
You would need to merge this in first :D https://github.com/neondatabase/neon/pull/7205/files#diff-9d2d8bce5039c6b425ede38184ed3cd8921e7019a6e49ecaa2ae24da075a6898R136-R146